src/test/java/com/kasperin/inventory_management/services/FruitAndVegeServiceImplTest.java

```diff
@@ -98,13 +103,16 @@ void getFruitAndVegeById() {
         when(fruitAndVegeRepository.findById(anyLong()))
                 .thenReturn(java.util.Optional.ofNullable(fruitAndVege));
 
-        Optional<FruitAndVege> result = fruitAndVegeService.findById(ID);
+        Optional<FruitAndVegeDTO> result = fruitAndVegeService.findById(ID);
 
         verify(fruitAndVegeRepository).findById(ID);
 
-        assertEquals(fruitAndVege, result.get());
-
+        FruitAndVegeDTO dto = result.get();
 
+        assertEquals(fruitAndVege.getBarcode(), dto.getBarcode());
+        assertEquals(fruitAndVege.getInStockQuantity(), dto.getInStockQuantity());
+        assertEquals(fruitAndVege.getName(), dto.getName());
+        assertEquals(fruitAndVege.getPrice(), dto.getPrice());
```

Comment:
Comparing fields individually since input and output objects aren't the same type